modules/dnn/src/dnn.cpp

```diff
@@ -1492,6 +1510,39 @@ struct Net::Impl : public detail::NetImplBase
         addLayerInput(ldInp, inNum, LayerPin(outLayerId, outNum));
         ldOut.requiredOutputs.insert(outNum);
         ldOut.consumers.push_back(LayerPin(inLayerId, outNum));
+
+        CV_LOG_VERBOSE(NULL, 0, "DNN: connect(" << outLayerId << ":" << outNum << " ==> " << inLayerId << ":" << inNum << ")");
+    }
+
+    int registerOutput(const std::string& outputName, int layerId, int outputPort)
+    {
+        int checkLayerId = getLayerId(outputName);
```

Comment:
GetLayerId operates on names of layer, instead of names of outputs of layers. Not sure yet how to fix this. We might want to revert https://github.com/opencv/opencv/pull/21490.

Reply:
> GetLayerId operates on names of layer

It is correct.

> instead of names of outputs of layers

I believe output names should not be used during the network construction (it uses internal "nodes" names).
Also output could not be replaced by layer (layers may have multiple output ports)

---

`registerOutput` function tries to create dummy "Identity" layer with "required name" (in the end of network construction).
`checkLayerId` is just a check to avoid duplicates, replacement and inconsistent state of network (dnn still doesn't support "outputs" properly and rely on "unconnected layers" - we need new API for that).

Did you observe error with message: `"Layer with name='%s' already exists id=%d (to be linked with %d:%d)"`?

Reply:
Yes, I did. It happened because the net contains layer with name X, followed by a layer with output named X, hence the error: we mix up layer name with layer 's output name.